src/renderer/entities/transactionBuilder/lib/compound-wallet.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  CallBuilder,
+  TransactionBuilder,
+  TransactionVisitor
+} from "@entities/transactionBuilder/model/transaction-builder";
+import {AccountInWallet, Wallet} from "@shared/core/types/wallet";
+import {Account} from "@shared/core";
+import {Exception} from "@zxing/library";
+import {ApiPromise} from "@polkadot/api";
+import {BaseTxInfo, OptionsWithMeta, UnsignedTransaction} from "@substrate/txwrapper-polkadot";
+import {SubmittableExtrinsic} from "@polkadot/api/types";
+
+export class CompoundWalletTransactionBuilder implements TransactionBuilder {
+
+  effectiveCallBuilder: CallBuilder;
+
+  readonly wallet: Wallet
+  readonly shards: Account[]
```

Comment:
We have a `ShardAccount` for such case

Reply:
Yeah saw it too. However I was not quite sure whether I can safely cast Account to ShardAccount in all cases  CompoundWalletTransactionBuilder is used